hedera-node/src/main/java/com/hedera/services/contracts/operation/HederaOperationUtil.java

```diff
@@ -201,4 +194,21 @@ public static long newContractExpiryIn(MessageFrame frame) {
 
 		return supplierExecution.get();
 	}
+
+
+	public static void cacheExistingValue(
+			final MessageFrame frame,
+			final Address address,
+			final Bytes32 key,
+			final UInt256 storageValue
+	) {
+		// Store the read if it is the first read for the slot/address
+		var updater = frame.getMessageFrameStack().getLast().getWorldUpdater().parentUpdater().orElse(null);
+		if (updater != null) {
```

Comment:
Out of curiosity, is it possible for this to be `null`? (Since message calls always begin with a stacked updater, it's not immediately obvious how... 🤔)

Reply:
Sonar is complaining because there is an `orElse(null)` at line 206.  The var could be the `Optional<Updater>` and the conditional be an `Optional.isPresent()`.  But that feels a tad bit obsessive since the safety is ensured by the current construction.